controller/instance_manager_controller.go

```diff
@@ -681,26 +681,36 @@ func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.I
 		return false, nil
 	}
 
-	allowDrainingNodeWithLastReplica, err := imc.ds.GetSettingAsBool(types.SettingNameAllowNodeDrainWithLastHealthyReplica)
+	nodeWithLastReplicaDrainingPolicy, err := imc.ds.GetSettingValueExisted(types.SettingNameNodeWithLastHealthyReplicaDrainPolicy)
```

Comment:
As I mentioned, let's keep the original code together with this new setting to make sure it's still supported until it's removed.

P.S. let's create a ticket for that setting removal and update the description of `SettingNameAllowNodeDrainWithLastHealthyReplica` deprecated. 

```go
	allowDrainingNodeWithLastReplica, err := imc.ds.GetSettingAsBool(types.SettingNameAllowNodeDrainWithLastHealthyReplica)
	if err != nil {
		return false, err
	}

    if allowDrainingNodeWithLastReplica {
        return true, nil
    }
	
    nodeWithLastReplicaDrainingPolicy, err := imc.ds.GetSettingValueExisted(types.SettingNameNodeWithLastHealthyReplicaDrainPolicy)
    if err != nil {
        return false, err
    }

	if nodeWithLastReplicaDrainingPolicy {
		return true, nil
	}
```

Reply:
Ok. I will keep both old setting and new setting. There will be a bit of confusion for the user as which setting will take precedence in the case:
* old setting blocks, new setting allows -> allow
* old setting allows, new setting blocks -> allow
But I guess it is fine temporary until we remove the old setting the confusion will go away.